src/account/UpgradeableModularAccount.sol

```diff
@@ -251,11 +253,13 @@ contract UpgradeableModularAccount is
 
         // Run any pre plugin exec specific to this caller and the `executeFromPluginExternal` selector
 
-        PostExecToRun[] memory postPermittedCallHooks =
-            _doPrePermittedCallHooks(IPluginExecutor.executeFromPluginExternal.selector, msg.sender);
+        PostExecToRun[] memory postPermittedCallHooks = _doPrePermittedCallHooks(
+            getPermittedCallKey(msg.sender, IPluginExecutor.executeFromPluginExternal.selector), msg.data
```

Comment:
Curious why we use `msg.data` instead of `data` here

Reply:
The hooks applied to `executeFromPluginExternal` are tied to the function selector on the account, rather than on the target contract (i.e. you can see in this line that it uses `IPluginExecutor.executeFromPluginExternal.selector`. ) So, when it is attached to that function, it makes sense to send the calldata for that function call to the hook, rather than the data for the inner call.

By doing this instead of using the inner `data` field, the hook can be certain that the incoming calldata will be in a predictable format, which is the abi-encoded call to `executeFromPluginExternal`. It will also get the value to be sent, and which address the account is making the call to, which would not be visible if we just sent `data` because it is a separate parameter.

Reply:
That makes sense, I was wondering why https://github.com/erc6900/reference-implementation/blob/main/src/account/UpgradeableModularAccount.sol#L194 and https://github.com/erc6900/reference-implementation/blob/main/src/account/UpgradeableModularAccount.sol#L202 still stick to `data`

Reply:
Yes, the difference with `executeFromPlugin` is that it emulates a call to the fallback, so it needs to use the inner `data` field. Example:
- Plugin A defines a function `foo(uint256)`
- Plugin B adds a pre-exec hook over `foo(...)` that reads the `uint256` from calldata and performs a check
- Plugin C requests permission to use `foo(...)` via `executeFromPlugin`.

When plugin C uses `executeFromPlugin` for `foo(...)`, plugin B should get the data for `foo(uint256)`, rather than `executeFromPlugin(abi.encodeCall(foo, (...))`. If it were to get the second format for the data, then every hook would need two mechanisms to decode calldata rather than just one. By sending the inner `data` rather than the entire `msg.data`, it makes it more straightforward to implement hooks.